samples/vboxwrapper/vbox_common.cpp

```diff
@@ -190,11 +200,12 @@ int VBOX_BASE::run(bool do_restore_snapshot) {
     vm_name = vm_master_name;
 
     // Check to see if the VM is already in a running state, if so, poweroff.
-    poll(false);
     if (online) {
```

Comment:
Simply putting the poll back doesn't do the trick because in this point we have no VM log to parse so it always fails. Had to bring back the old poll function (poll2) as you suggested and use it at this point.